packages/grafana-data/src/transformations/matchers/nameMatcher.ts

```diff
@@ -40,8 +40,32 @@ const fieldNameMatcher: FieldMatcherInfo<string> = {
   defaultOptions: '',
 
   get: (name: string): FieldMatcher => {
+    let fallback: FieldMatcher | undefined = undefined;
+
+    // In an effor to support migrating to a consistent data contract, the
+    // naming conventions need to get normalized.  However many existing setups
+    // exist that would no longer match names if that changes.  This injects
+    // fallback logic when when the data frame has not type version specified
+    if (name === 'Value') {
+      fallback = (field: Field, frame: DataFrame) => {
+        return (
+          frame.meta?.typeVersion == null &&
```

Comment:
perhaps this should be the other way around 🤔 or keep typeVersion out of this entirely?

Reply:
Similar to https://github.com/grafana/grafana/pull/65656, I think we should keep typeVersion out of this. Can we also check for 
`field.labels?.__name__ === field.name`?

Reply:
that seems better -- we should do that instead of matching all numbers!

Also `Value` should use the constant